### ISSUE TITLE
`(Friday + March) / April === 2.0`

### ISSUE THREAD

**Issue by LilithHafner** [JuliaLang/julia#51357]
This is too permissive.

**Comment by Seelengrab** [comment]
This is because none of the months have a proper type - they are literally numeric constants:

https://github.com/JuliaLang/julia/blob/master/stdlib/Dates/src/query.jl#L259-L270

**Comment by jacobusmmsmit** [comment]
I can't think of any languages that have months and weeks be treated as integers by default.
Is there a precedent for this in a language someone knows? Is it useful in any applications?

**Comment by timholy** [comment]
Presumably they should be `@enum`s

**Comment by llbit** [comment]
> I can't think of any languages that have months and weeks be treated as integers by default. Is there a precedent for this in a language someone knows? Is it useful in any applications?

It could be useful for iterating over a range of months, for example `August:December`

**Comment by Seelengrab** [comment]
That can easily be accomodated by another approach as well. Ranges are not exclusive to numbers; `'a':'z'` works too, for example, which has an analogous situation.

**Comment by KristofferC** [comment]
I don't think fixing this has to be a 2.0 thing.

**Comment by jacobusmmsmit** [comment]
How about making Dates an upgradeable stdlib?

**Comment by JeffBezanson** [comment]
I don't think this can be considered a bug --- Dates exports these constants and the design, clearly, was to make them just integers. You can certainly disagree with that design but changing it now *would* cause bugs.

**Comment by Seelengrab** [comment]
Right - hence the requirement to do this in a major version bump of the stdlib (which need not coincide with a major version bump of Julia, if I'm not mistaken).

**Comment by simonbyrne** [comment]
See #19210 (and #19221, which was folded into it)

**Comment by Seelengrab** [comment]
A bit depressing that this has been open for numerous breaking releases now :grimacing:

**Comment by KristofferC** [comment]
Dates could be moved to an upgradable stdlib and there could be a breaking version made there maybe?

**Comment by jacobusmmsmit** [comment]
The only potential problem I see with upgradeable stdlibs is backporting bugfixes and maintaining the non-breaking Dates package. Would everyone who just uses Julia 1.X without touching Dates be left bugfixless?

**Comment by Seelengrab** [comment]
Has seen Dates that much movement/fixes in the past that this is a concern? Most Dates stuff seems recent and about the core design of the package, not just related to easily fixable bugs.
